### PR TITLE
Initial commit of Sommelier Chain Genesis

### PR DESCRIPTION
**Pull request by sommelierltd**
This is a first draft of the sommelier chain genesis.

It currently doesn't have any vesting in the genesis state. This will need to be added before the PR can be finalized.

Please review for errors.

### CODE REVIEW

**Comment by zmanian** [comment]
@fedekunze can you help add usomm denom info to the genesis file?

File: genesis.json [hunk 1]
```diff
@@ -0,0 +1,560 @@
+{
+  "genesis_time": "2021-01-19T18:53:31.685Z",
+  "chain_id": "sommelier",
```
Comment (fedekunze) [inline genesis.json]: this needs to have the epoch number for IBC compat
```suggestion
  "chain_id": "sommelier-1",
```

File: genesis.json [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+      "unbonding_delegations": [],
+      "redelegations": [],
+      "exported": false
+    },
+    "transfer": {
+      "port_id": "transfer",
+      "denom_traces": [],
+      "params": {
+        "send_enabled": false,
+        "receive_enabled": false
```
Comment (fedekunze) [inline genesis.json]: should we enable IBC at launch?

**Comment by taariq** [comment]
Yessss!

File: genesis.json (hunk 2 above)
Comment (zmanian) [reply genesis.json]: I suggest we not enable IBC until we have vesting enabled.

I've been watching the periodic vesting fixes in the SDK.

I think we will want to get a version of this up and then do vesting + IBC in somm 2

File: genesis.json [hunk 3]
```diff
@@ -0,0 +1,7122 @@
+{
+  "genesis_time": "2021-01-19T18:53:31.685Z",
+  "chain_id": "sommelier-1",
+  "initial_height": "1",
+  "consensus_params": {
+    "block": {
+      "max_bytes": "22020096",
```
Comment (jackzampolin) [inline genesis.json]: This is large. Prob a good idea tho. Any reasoning on the number?

File: genesis.json [hunk 4]
```diff
@@ -0,0 +1,7122 @@
...
+      "max_gas": "-1",
```
Comment (jackzampolin) [inline genesis.json]: Do we maybe want to set some limit here?

File: genesis.json [hunk 5]
```diff
@@ -0,0 +1,7122 @@
...
+          "permissions": [
+            "basic"
+          ]
+        }
+      ]
+    },
+    "bank": {
+      "params": {
+        "send_enabled": [],
+        "default_send_enabled": false
```
Comment (jackzampolin) [inline genesis.json]: Ack send disabled

File: genesis.json [hunk 6]
```diff
@@ -0,0 +1,7122 @@
...
+        "min_deposit": [
+          {
+            "denom": "usomm",
+            "amount": "10000000"
+          }
+        ],
+        "max_deposit_period": "172800s"
+      },
+      "voting_params": {
+        "voting_period": "172800s"
```
Comment (jackzampolin) [inline genesis.json:7177]: 2 day voting period 👍

File: genesis.json [hunk 7]
```diff
@@ -0,0 +1,7122 @@
...
+      },
+      "tally_params": {
+        "quorum": "0.005000000000000000",
```
Comment (jackzampolin) [inline genesis.json]: What about 5-10% Also open to .5%.

File: genesis.json [hunk 8]
```diff
@@ -0,0 +1,7122 @@
...
+        "threshold": "0.500000000000000000",
+        "veto_threshold": "0.334000000000000000"
+      }
+    },
+    "gravity": {
+      "params": {
+        "gravity_id": "defaultgravityid",
```
Comment (jackzampolin) [inline genesis.json]: `sommelier`

File: genesis.json [hunk 9]
```diff
@@ -0,0 +1,7122 @@
...
+        "contract_source_hash": "",
```
Comment (jackzampolin) [inline genesis.json:7188]: I feel like putting this in there would be a nice thing to do. I don't think its wired up anywhere, but 🤷‍♂️

File: genesis.json [hunk 10]
```diff
@@ -0,0 +1,7122 @@
...
+        "bridge_ethereum_address": "0x0000000000000000000000000000000000000000",
+        "bridge_chain_id": "0",
```
Comment (jackzampolin) [inline genesis.json]: `1`

File: genesis.json [hunk 11]
```diff
@@ -0,0 +1,7122 @@
...
+        "signed_signer_set_txs_window": "10000",
+        "signed_batches_window": "10000",
+        "ethereum_signatures_window": "10000",
+        "target_eth_tx_timeout": "43200000",
+        "average_block_time": "5000",
+        "average_ethereum_block_time": "15000",
+        "slash_fraction_signer_set_tx": "0.001000000000000000",
+        "slash_fraction_batch": "0.001000000000000000",
+        "slash_fraction_ethereum_signature": "0.001000000000000000",
+        "slash_fraction_conflicting_ethereum_signature": "0.001000000000000000",
```
Comment (jackzampolin) [inline genesis.json]: Did we want to disable slashing for this stuff?

File: genesis.json [hunk 12]
```diff
@@ -0,0 +1,7122 @@
...
+        "inflation_max": "0.000000000000000000",
+        "inflation_min": "0.00000000000000000",
+        "goal_bonded": "0.670000000000000000",
+        "blocks_per_year": "6311520"
+      }
+    },
+    "params": null,
+    "slashing": {
+      "params": {
+        "signed_blocks_window": "100",
```
Comment (jackzampolin) [inline genesis.json]: These slashing settings say that any more than ~5 minutes consecutive minutes of downtime will produce jailing for a day but no slashing. May I propose something closer to ~50 minutes consecutive downtime producing jailing for a day? Could be convinced to the side of 5 min too.

File: genesis.json [hunk 13]
```diff
@@ -0,0 +1,7122 @@
...
+        "downtime_jail_duration": "86400s",
+        "slash_fraction_double_sign": "0.000000000000000000",
+        "slash_fraction_downtime": "0.000000000000000000"
+      },
+      "signing_infos": [],
+      "missed_blocks": []
+    },
+    "staking": {
+      "params": {
+        "unbonding_time": "1814400s",
```
Comment (jackzampolin) [inline genesis.json]: `1209600s`? (14 days) or `2419200s` (28 days)? We could also go for something longer.

File: genesis.json [hunk 14]
```diff
@@ -0,0 +1,7122 @@
...
+        "max_validators": 100,
```
Comment (jackzampolin) [inline genesis.json]: What about 50?

File: genesis.json (hunk 2 above)
Comment (jackzampolin) [reply genesis.json]: Bump this comment: Do we want IBC enabled?

File: genesis.json [hunk 15]
```diff
@@ -7094,8 +7094,8 @@
     },
     "staking": {
       "params": {
-        "unbonding_time": "1814400s",
-        "max_validators": 100,
+        "unbonding_time": "1209600s",
```
Comment (jackzampolin) [inline genesis.json]: Like this a lot.

File: genesis.json [hunk 16]
```diff
@@ -0,0 +1,7122 @@
...
+        "blocks_per_year": "6311520"
+      }
+    },
+    "params": null,
+    "slashing": {
+      "params": {
+        "signed_blocks_window": "5000",
+        "min_signed_per_window": "0.050000000000000000",
+        "downtime_jail_duration": "8640s",
+        "slash_fraction_double_sign": "0.000000000000000000",
```
Comment (mvid) [inline genesis.json:7258]: Do we want validators to be allowed to double sign?

Comment (jackzampolin) [reply genesis.json:7258]: I think we are going to want 0 slashing on this initial network. We will add these features back in via gov at a later time

**Pull request merged**